googledest/users.go

```diff
@@ -181,44 +189,89 @@ func (g *GoogleUsers) ApplyChangeSet(
 func newUserForUpdate(person personnel_sync.Person, oldUser admin.User) (admin.User, error) {
 	user := admin.User{}
 	var err error
-
-	newName := admin.UserName{
-		GivenName:  person.Attributes["givenName"],
-		FamilyName: person.Attributes["familyName"],
-	}
-	user.Name = &newName
-
-	if person.Attributes["id"] != "" {
-		user.ExternalIds, err = updateIDs(person.Attributes["id"], oldUser.ExternalIds)
-		if err != nil {
-			return admin.User{}, err
+	var organization *admin.UserOrganization
+
+	for key, val := range person.Attributes {
+		switch key {
+		case "givenName":
+			if user.Name == nil {
+				user.Name = &admin.UserName{GivenName: val}
+			} else {
+				user.Name.GivenName = val
+			}
+
+		case "familyName":
+			if user.Name == nil {
+				user.Name = &admin.UserName{FamilyName: val}
+			} else {
+				user.Name.FamilyName = val
+			}
+
+		case "id":
+			user.ExternalIds, err = updateIDs(val, oldUser.ExternalIds)
+			if err != nil {
+				return admin.User{}, err
+			}
+
+		case "area":
+			user.Locations, err = updateLocations(val, oldUser.Locations)
+			if err != nil {
+				return admin.User{}, err
+			}
+
+		case "costCenter":
+			if organization == nil {
+				organization = &admin.UserOrganization{CostCenter: val}
+			} else {
+				organization.CostCenter = val
+			}
+
+		case "department":
+			if organization == nil {
+				organization = &admin.UserOrganization{Department: val}
+			} else {
+				organization.Department = val
+			}
+
+		case "title":
+			if organization == nil {
+				organization = &admin.UserOrganization{Title: val}
+			} else {
+				organization.Title = val
+			}
+
+		case "phone":
+			user.Phones, err = updatePhones(val, oldUser.Phones)
+			if err != nil {
+				return admin.User{}, err
+			}
+
+		case "manager":
+			user.Relations, err = updateRelations(val, oldUser.Relations)
+			if err != nil {
+				return admin.User{}, err
+			}
+
+		default:
+			keys := strings.SplitN(key, ".", 2)
+			if len(keys) < 2 {
+				continue
+			}
+
+			j, err := json.Marshal(&map[string]string{keys[1]: val})
+			if err != nil {
+				return admin.User{}, fmt.Errorf("error marshaling location, %s", err)
```

Comment:
Is the error message correct on this (i.e. `location` vs. `custom schema` or something)?

Reply:
You're right. Thanks.